### PR TITLE
runtime: io_signature: improve exception messages

### LINKED ISSUE #6758: io_signature exceptions are really bad
### What happened?

Got a user reporting an error when creating an io_signature from Python. I need to read code to learn what

     ValueError: gr::io_signature(3)

means. 

### System Information

3.11git

### GNU Radio Version

3.11-git (main)

### Specific Version

0.0.0.0

### Steps to Reproduce the Problem

```Python
from gnuradio import gr
gr.io_signature.makev(10, 10, [gr.sizeof_gr_complex*1, gr.sizeof_gr_complex*1, gr.sizeof_float*1, gr.sizeof_float*1, gr.sizeof_float*1, 0*1, gr.sizeof_float*1, gr.sizeof_float*1, gr.sizeof_float*1, gr.sizeof_float*1])
```

### Relevant log output

```shell
ValueError                                Traceback (most recent call last)
Cell In[4], line 1
----> 1 gr.io_signature.makev(10, 10, [gr.sizeof_gr_complex*1, gr.sizeof_gr_complex*1, gr.sizeof_float*1, gr.sizeof_float*1, gr.sizeof_float*1, 0*1, gr.sizeof_float*1, gr.sizeof_float*1, gr.sizeof_float*1, gr.sizeof_float*1])

ValueError: gr::io_signature(3)
```

### Remarks

- Error messages for this could actually be self-explanatory. 
- I'm not sure whether changing an exception string would be considered an API break
- Using `fmt`, as available through spdlog anyways, one could have nice error mesages that explicitly say what's wrong
-  Good first issue for someone who knows how to compile GNU Radio!

### PR DESCRIPTION
**Pull request by marcusmueller**
minor code style improvements, removal of unused config.h

## Description

The current messages you get in Python land if you specify invalid arguments for `io_signature` creation are cryptic. That's not any better in C++ land, but chances are, C++ users might have an easier time finding the GR source code to help themselves.

## Related Issue

Fixes #6758

## Which blocks/areas does this affect?

System-wide, but only in terms of verbage of exception messages. I do hope nobody checked for string equality in exception handlers.

## Testing Done

All tests still run through

## Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md).
- [x] I have squashed my commits to have one significant change per commit. 
- [x] I [have signed my commits before making this PR](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md#dco-signed)
- [x] My code follows the code style of this project. See [GREP1.md](https://github.com/gnuradio/greps/blob/main/grep-0001-coding-guidelines.md).
- [x] I have updated [the documentation](https://wiki.gnuradio.org/index.php/Main_Page#Documentation) where necessary.
- [ ] I have added tests to cover my changes, and all previous tests pass.

### CODE REVIEW

File: gnuradio-runtime/lib/io_signature.cc [hunk 1]
```diff
@@ -90,28 +90,43 @@ io_signature::io_signature(int min_streams,
...
 
 io_signature::~io_signature() {}
 
-int io_signature::sizeof_stream_item(int _index) const
+int io_signature::sizeof_stream_item(int index) const
 {
-    if (_index < 0)
-        throw std::invalid_argument("gr::io_signature::sizeof_stream_item");
+    if (index < 0)
+        throw std::invalid_argument(fmt::format("Index < 0: {:d}", index));
```
Comment (mormj) [inline gnuradio-runtime/lib/io_signature.cc:124]: Should this error message be more descriptive, such as having "sizeof_stream_item" in the message?

File: gnuradio-runtime/lib/io_signature.cc [hunk 2]
```diff
@@ -90,28 +90,43 @@ io_signature::io_signature(int min_streams,
...
     if (sizeof_stream_items.empty()) {
-        throw std::invalid_argument("gr::io_signature(2)");
+        throw std::invalid_argument("Item size vector cannot be empty");
     }
-
-    for (size_t i = 0; i < sizeof_stream_items.size(); i++) {
-        if (max_streams != 0 && sizeof_stream_items[i] < 1)
-            throw std::invalid_argument("gr::io_signature(3)");
+    if (max_streams) {
+        decltype(sizeof_stream_items.size()) index = 0;
```
Comment (willcode) [inline gnuradio-runtime/lib/io_signature.cc:108]: Are there any cases where `size_t` wouldn't work here ... just to keep the syntax more readable?

File: gnuradio-runtime/lib/io_signature.cc (hunk 1 above)
Comment (marcusmueller) [reply gnuradio-runtime/lib/io_signature.cc:124]: agreed!

File: gnuradio-runtime/lib/io_signature.cc (hunk 2 above)
Comment (marcusmueller) [reply gnuradio-runtime/lib/io_signature.cc:108]: yeah, you're right!

File: gnuradio-runtime/lib/io_signature.cc [hunk 3]
```diff
@@ -90,28 +90,44 @@ io_signature::io_signature(int min_streams,
...
-        throw std::invalid_argument("gr::io_signature(1)");
-
+    if (min_streams < 0) {
+        throw std::invalid_argument(
+            fmt::format("Minimum number of streams must be >= 0, is {:d}", min_streams));
+    }
+    if (max_streams != IO_INFINITE && max_streams < min_streams) {
+        throw std::invalid_argument(
+            fmt::format("Maximum number of streams ({:d}) needs to be at least as "
+                        "large as minimal number of streams ({:d})",
```
Comment (willcode) [inline gnuradio-runtime/lib/io_signature.cc:100]: minimum

**Review by willcode — APPROVED** [review]

**Comment by marcusmueller** [comment]
superseded by #6760

**Pull request closed without merging**